src/Bicep.Core/Syntax/TestDeclarationSyntax.cs

```diff
@@ -53,9 +53,31 @@ this.Value switch
                 // blocked by assert in the constructor
                 _ => throw new NotImplementedException($"Unexpected type of test value '{this.Value.GetType().Name}'.")
             };
+        
+        public ObjectSyntax? TryGetParameters(ObjectPropertySyntax property)=>
+            property.Value switch
+            {
+                ObjectSyntax @object => @object,
+                SkippedTriviaSyntax => null,
+
+                // blocked by assert in the constructor
+                _ => throw new NotImplementedException($"Unexpected type of params value '{this.Value.GetType().Name}'.")
+            };
+        
 
         public ObjectSyntax GetBody() =>
             this.TryGetBody() ?? throw new InvalidOperationException($"A valid test body is not available on this test due to errors. Use {nameof(TryGetBody)}() instead.");
 
+        public ObjectSyntax GetParameters(){
+            var body = this.GetBody();
+            foreach(var property in body.Properties){
+                if(property.TryGetKeyText() == "params"){
+                    return this.TryGetParameters(property) ??  throw new InvalidOperationException($"A valid parameters body is not available on this test due to errors. Use {nameof(TryGetParameters)}() instead."); ;
+                }
+            }
+            return body;
```

Comment:
```suggestion
            var body = this.GetBody();
            foreach (var property in body.Properties) {
                if (property.TryGetKeyText() == "params" && property.Value is ObjectSyntax paramsObject) 
                {
                    return paramsObject;
                }
            }
            return null;
```

You would be able to remove some of the unexpected state handling if the method returns a nullable `ObjectSyntax?` value.